lib/improver/tests/ensemble_calibration/ensemble_calibration/test_EnsembleCalibration.py

```diff
@@ -72,9 +72,12 @@ class Test_process(IrisTest):
         ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
     def setUp(self):
         """Set up temperature and wind speed cubes for testing."""
+        # Note: test_temperature_realizations_data_check produces ~0.5K
+        # different results when the temperature forecast cube is float32
+        # below. A bug?
```

Comment:
If we alter the `dtype=np.float64` to `dtype=np.float32` (or remove the `dtype` keyword altogether) then we get a unit test failure with:

```
mismatch 100.0%)
 x: array([[ 231.1493,  242.3992,  253.6492],
       [ 264.8991,  276.149 ,  287.399 ],
       [ 298.649 ,  309.8989,  321.1488]], dtype=float32)
 y: array([[ 230.7225,  241.9444,  253.1663],
       [ 264.3882,  275.6102,  286.8321],
       [ 298.054 ,  309.2759,  320.4979]], dtype=float32)
```

Reply:
```
"/opt/improver/lib/improver/tests/ensemble_calibration/ensemble_calibration/test_EnsembleCalibration.py", line 267, in test_temperature_realizations_data_check
    decimal=4)
```

Reply:
I've had a look at this, and I think this is all arising from the difference in the initial guess, resulting from the change in precision, that is being created when using the ordinary least squares regression to generate the initial guess for the ensemble calibration coefficients.

The initial guess coefficients listed below can be understood as:
Intercept | Ensemble member 1 gradient | Ensemble member 2 gradient | Ensemble member 3 gradient

For example, the previous version created the following values for the initial guess coefficients in the `test_temperature_realizations_data_check` test:
```0.084745762711864389, 0.05084745762711841, 0.38983050847457623, 0.559322033898305```
After iteration, these became:
```0.10456126  0.10277997  0.66682032  0.7364042```

However, using float32 input, the initial guess is:
```0.99631459 -0.00170303  1.00326633 -0.00156335``` 
After iteration, these became:
```1.00036648 -0.00196032  0.99999076 -0.00314605```

As can be seen, the iteration isn't having that much effect with the result being critical upon the initial guess. This is likely because the unit tests are using a small amount of data, so there isn't a lot of refinement that the iteration can do. Therefore, I think the changes resulting from the conversion to float32 are alright in this situation, because it is only arising because of the fact that the unit tests are using a small amount of idealised data and there are multiple potential values for the coefficients that could satisfy the ordinary least squares regression.

Reply:
Thanks Gavin!

Reply:
I've now pushed up a change that uses float32 here, with new numbers in the unit test KGO arrays.